artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/CoreAmqpConverter.java

```diff
@@ -456,7 +450,7 @@ private static Section convertBody(ServerJMSMessage message, Map<Symbol, Object>
             while (true) {
                list.add(m.readObject());
             }
-         } catch (MessageEOFException e) {
+         } catch (Exception e) {
```

Comment:
This should only be catching/ doing this for EOF exceptions not for all!

Reply:
There's nothing else that could happen here...

I don't see the need to duplicate the entire JMS Exception just for these...